benchmarks/graphics_pipeline/main.cpp

```diff
@@ -137,10 +171,11 @@ class ProjApp
     grfx::DescriptorSetLayoutPtr                                  mSetLayout;
     std::array<grfx::ShaderModulePtr, kAvailableVsShaders.size()> mVsShaders;
     std::array<grfx::ShaderModulePtr, kAvailablePsShaders.size()> mPsShaders;
-    PerspCamera                                                   mCamera;
+    FreeCamera                                                    mCamera;
     float3                                                        mLightPosition = float3(10, 100, 10);
     std::array<Scene, kAvailableScenes.size()>                    mScenes;
     size_t                                                        mCurrentSceneIndex;
+    std::set<ppx::KeyCode>                                        mPressedKeys;
```

Comment:
an std::set is an RB-tree, with log complexity for insert.
What you want is probably an unordered_set.

Also, since the we have like 30 keycodes, maybe this could be a simple std::array with a boolean in each cell (`array[KEY_SPACE] = true`) ?